provisioners/src/main/java/org/jboss/intersmash/provision/operator/OpenShiftAIOperatorProvisioner.java

```diff
@@ -105,16 +109,21 @@ public void undeploy() {
 				.level(Level.DEBUG)
 				.waitFor();
 
-		if (getApplication().getDSCInitialization() != null) {
-			final String appName = getApplication().getName();
-			deletionDetails = dscInitializationClient().withName(appName).delete();
-			deleted = deletionDetails.stream().allMatch(d -> d.getCauses().isEmpty());
-			if (!deleted) {
-				log.warn("Wasn't able to remove the 'DSCInitialization' resources created for '{}' instance!",
-						appName);
-			}
-			new SimpleWaiter(() -> dscInitializationClient().list().getItems().isEmpty()).level(Level.DEBUG).waitFor();
+		deletionDetails = dscInitializationClient().delete();
+		deleted = deletionDetails.stream().allMatch(d -> d.getCauses().isEmpty());
+		if (!deleted) {
+			log.warn("Wasn't able to remove the 'DSCInitialization' resources created for '{}' instance!",
```

Comment:
have you checked if this warning is logged or not? in my tests the resource can't be deleted because of the **finalizer**:
```
$ oc get DSCInitialization/default-dsci --all-namespaces 
NAME           AGE    PHASE         CREATED AT
default-dsci   3h9m   Progressing   2025-12-03T08:13:30Z
```
we might have to patch and remove **finalizer**s before deleting...

Reply:
Not really, patching finalizers is a last resort choice and only done manually when something's gone wrong previously. Here there's no need to do so since the workflow seems to be clean now.